model/dns_zone/dns_zone.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require "zlib"
+require_relative "../../model"
+
+class DnsZone < Sequel::Model
+  many_to_one :project
+  one_to_many :resource_records, class: :DnsZoneResourceRecord
+  one_to_many :active_billing_records, class: :BillingRecord, key: :resource_id, conditions: {Sequel.function(:upper, :span) => nil}
+
+  include ResourceMethods
+  include Authorization::HyperTagMethods
+  include Authorization::TaggableMethods
+
+  def hyper_tag_name(project)
+    "project/#{project.ubid}/location/#{location}/dns_zone/#{name}"
```

Comment:
Do we use underscores in urls? they're somewhat odd stylistically. But I remember @enescakir had some triangulation with how they show up in the authorization language that may make it the least worst option, or something. I forget. @enescakir ?

Reply:
Underscores are not valid in the domains, so we wouldn't allow them in the dns zone name.

Reply:
If underscores cause problem, we can use ubid instead of name.

Reply:
Changed it to ubid.